### ISSUE TITLE
Join in Reserve and Customs have a low FPS but other map not.

### ISSUE THREAD

**Issue by kaguyaAki** [paulov-t/SIT.Core#192]
As the say with title, you join in other player's room ,when the map is Reserve or Customs, your FPS will became very low than other map. Even if the worst  map “street of Tarkov”，also have the high FPS than Reserve and Customs. In my PC Reserve only less than 15FPS and Customs less than 30. But  Street of Tarkov will have 90+ FPS .

**Comment by KamiliaBlow** [comment]
I can confirm this problem. And it occurs only with the second player. The 3 player has no FPS slippage. For some reason, it affects only player 2. And the FPS keeps dropping over time.

My spec:
cpu: AMD Ryzen 7 3700x
gpu: RTX 3080
ram: 32gb ddr4

Connection type: Radmin VPN (full free analogue Hamachi)
I don't use Direct IP because the server just won't start and gives an error that port 6969 is already in use.

![image](https://github.com/paulov-t/SIT.Core/assets/25077548/8b60abd8-68ea-4f99-a585-e7612ee96196)

**Comment by KamiliaBlow** [comment]
The same error was with the people with whom I tested the game, even me, if I connect as a second player FPS dropped. Because of what drops FPS is unknown, but it happens to my players every time. Since I mostly test alone, I gave the example of my clients.

> You cannot rely on running a second instance of a game to be your source of truth.

**Comment by TehFl0w** [comment]
I had a similar experience so far, as a client my fps usually drop over time no matter the map (although customs is noticeably worse and it does seem to only affect the second player) but rarely to an unplayable level. 

My specs are on the higher end so hardware shouldn't be the issue.

Our server runs on a remote machine and connection-wise everything seems fine, pings around 5-30ms.

I sometimes experience a heavy stutter after someone gets shot and/or dies with game performance and ping being slightly worse afterwards, maybe this could be related? Also in those cases the ping-display in the upper left corner displays "BAD PERFORMANCE" if that could help narrow it down

I'll keep an eye out for any further information or patterns

**Comment by Lega955** [comment]
Same situation for us, the host has significantly higher FPS than the client. Today, we decided to switch to a direct connection instead of using ZeroTier, and it seems that the client's FPS has improved. Previously, when using ZeroTier, the clients were experiencing around 25 FPS, while the host consistently had 60-70 FPS. However, with the direct connection, the clients now have an average of 60 FPS, which is completely playable. We also attempted to rotate the hosts, but the results were similar.


@KamiliaBlow, I faced a similar issue with port 6969 already being used. I resolved it by using my local IPv4 address instead of the IP displayed on myip.com (Public IP) in the Aki_data/Server/configs/HTTP.json file. However, the clients will still need your public IP address to join, hope this helps!

**Comment by kaguyaAki** [comment]
There is no intention of complaining because except for joining these two maps, other maps works very well. I'm just curious why the FPS of maps like "streets of tarkov" hasn't decreased much, but instead Reserve and Customs have so low FPS. Of course, this only happens to players who join, first  player who start a maps is well than online game.
I use a app like Hamachi to create the server. According to other players, perhaps this is the problem?


My spec:
cpu: AMD Ryzen 7 3700x
gpu: RTX 2070 super
ram: 64gb ddr4

> PC Specs? Connection speed? Connection method?
> 
> Reason I ask this. I have 2 people with lower-mid end PCs, connecting via Direct IP and they are not complaining at all. In fact, they all compliment how much it runs better than Live.

**Comment by nowblank** [comment]
@paulov-t   @TehFl0w 

on local server connection my fps seems to drop around 30fps after encountering bots on shoreline map..  

PC spec:
cpu: i7 8700k
gpu: 7900 xtx
ram: 16gb ddr4

**Comment by mrbe3yh4uk** [comment]
Same situation for us, we using direct connection

**Comment by Lega955** [comment]
> 
The paging file size trick can be effective, but it depends on your current setup. It is recommended to place the paging file on your fastest drive. However, increasing the size above 30000MB may not provide significant benefits based on my own tests, although individual results may vary. Keep in mind that the impact of the paging file size also depends on the amount of RAM you have. If you already have 32GB DDR4 RAM, adjusting the paging file size may not have a noticeable impact.

**Comment by eagleeyes117** [comment]
I agree with that the noticeable difference was on a laptop with 16g of ram on my built pc 32g it doesn't do much i don't know what buddy had for a set up but i thought i would mention it anyway

Also i really appreciate your work paulov-t

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by derbismarck** [comment]
We've encountered this one, it seems to be related to AI replication in some way. Two clients, <30ms ping, the host on Customs usually has a steady framerate throughout. We adjusted our spawn waves to be delayed five minutes, so we're able to observe that on the client the FPS is quite high (as expected) until AI start spawning. After that five minute delay has elapsed, we observe that each time an AI spawns (whether Scav or PMC) the client machine sees about a 5FPS drop, which will then recover slightly. Depending on how you have AI counts configured and the specs on the client machine, the client FPS can get quite low. In our example, for Customs:

SPT-Aki dedicated server specs:
    - i9-9900k
    - 1gbit up/down
    - SSD
    - 64GB RAM (lol)

~20 AI at peak:
Player #1: 10700k, 16GB RAM, M.2 SSD, GTX 1070 playing at 4K
    - When hosting: 30 to 50 FPS, depending on location; usually closer to ~45.
    - When client: ~45FPS before AI spawn, 20 to 25 FPS after, depending on AI number; degrades over time and doesn't recover even if AI are killed
 
Player #2: 6700k, 32GB RAM, SATA SSD, RTX 2060 playing at 1080p
    - When hosting: 60FPS+, depending on location
    - When client: 60FPS+ before AI spawn, 15 to 20 FPS after, depending on AI number; degrades over time and doesn't recover even if AI are killed


In contrast, on Factory we observe no such performance hits on clients when AI spawn in. Woods is fine, as is Shoreline, as is Streets.

This issue seems to have an outsized impact on the following maps for us, for whatever reason. Maybe it's that these maps are already taxing in some way and the replication/serialization/de-serialization routines have more of an impact because of that?

- Interchange
- Customs


We have not tried playing on Reserve at all, and have not played on Labs with more than one person.

**Comment by jank04** [comment]
I can confirm this behaviour. The FPS from the second player is lower than usual.
This can be seen at least on customs, labs and especially reserve. My friend had 15 FPS, with a 3090.

(virtual)Server specs:
Windows Server 2022 Standard
Intel Xeon CPU E5-2650 0 (x8)
16 GB DDR3-ECC RAM

CPU usage peaks at 20%. "Connection Mode" is dyn dns with port forwarding. Me as first player is connected via LAN, second player is connected via dyn dns and WAN.

**Comment by derbismarck** [comment]
+1 on Reserve being the worst performing map for player clients. With 12 bots, we have no player client who consistently has more than 15 FPS. Player host as usual will have 30FPS or more.
